### PR TITLE
add TTS support

### PR DESCRIPTION
**Pull request by hypevhs**
Here's how you'd set it up in Arch
```zsh
yay -S speech-dispatcher
cat << EOF > ~/.config/systemd/user/speech-dispatcherd.service
[Unit]
Description=Speech-Dispatcher, common interface to speech synthesizers

[Service]
Type=forking
ExecStart=/usr/bin/speech-dispatcher -d -t 0
ExecReload=/bin/kill -HUP \$MAINPID

[Install]
WantedBy=default.target
EOF
systemctl --user daemon-reload
systemctl enable --now --user speech-dispatcherd.service
```
Install [Pied](https://github.com/Elleo/pied/) Flatpak and select a voice
Launch Vesktop
Enable VcNarrator plugin and play the test clips

### CODE REVIEW

File: dev.vencord.Vesktop.yml [hunk 1]
```diff
@@ -71,3 +72,49 @@ modules:
...
+              - /lib/pkgconfig
+              - /share/doc
+
+      # # From inside the sandbox, this may be helpful to debug the connection to the host's speech-dispatcher unix socket
+      # # printf 'speak\r\nhello\r\n.\r\n' | /app/bin/socat - "$XDG_RUNTIME_DIR/speech-dispatcher/speechd.sock"
+      # - name: socat
+      #   sources:
+      #     - type: archive
+      #       url: http://www.dest-unreach.org/socat/download/socat-1.8.0.0.tar.gz
+      #       sha512: edf459a9f1907a14025e13b3101ad29787f9a72795cffcd00017ce98847562884db29a95b9ae478a6a50868137548b142947c43fb18e975eb5853a763c42902c
```
Comment (lewisakura) [inline dev.vencord.Vesktop.yml]: Don't think we need to leave this here if we aren't using it

**Review by lewisakura — DISMISSED** [review]

File: dev.vencord.Vesktop.yml (hunk 1 above)
Comment (hypevhs) [reply dev.vencord.Vesktop.yml]: I removed it

**Comment by hypevhs** [comment]
@lewisakura Can we take another look at this?

File: dev.vencord.Vesktop.yml [hunk 2]
```diff
@@ -71,3 +72,41 @@ modules:
         x-checker-data:
           type: electron-updater
           url: https://github.com/Vencord/Vesktop/releases/latest/download/latest-linux-arm64.yml
+
+    modules:
+      # borrowed and tweaked from the net.lutris.Lutris manifest and the Arch PKGBUILD
```
Comment (Vendicated) [inline dev.vencord.Vesktop.yml]: could you please also include a permalink to the source(s)?

File: dev.vencord.Vesktop.yml [hunk 3]
```diff
@@ -71,3 +72,41 @@ modules:
...
+        cleanup:
+          - "*.la"
+          - "*.a"
+          - /include
+        sources:
+          - type: archive
+            url: https://github.com/brailcom/speechd/releases/download/0.11.5/speech-dispatcher-0.11.5.tar.gz
+            sha512: d6d880bce0ae5bc2a5d519ef7740c689ae8b4b0bb658379762810e4beae3e465a429fbe19f7c490e89db0ea6a36aedd4b2287ac9251b90059b5c2cb3c0dd8a28
+        modules:
+          - name: dotconf
```
Comment (Vendicated) [inline dev.vencord.Vesktop.yml]: what exactly is this needed for or used by?

**Review by Vendicated — COMMENTED** [review]
thanks for your contribution!

i am honestly not sure this is the right place to add this... wouldn't it be more appropriate to make this addition in the electron base? that way all electron apps can benefit from it

**Comment by hypevhs** [comment]
I moved the modules stuff to the electron baseapp and rebuilt both, and it still gets picked up by Chromium correctly. I'll open a PR with them later, but if they turn me away I'm coming right back here :P
With Vencord plugins being broken at the moment, though, I'll hold off.

File: dev.vencord.Vesktop.yml (hunk 3 above)
Comment (hypevhs) [reply dev.vencord.Vesktop.yml]: dotconf is a dependency of speech-dispatcher. It provides utility functions to parse config files. It's really only used in the server, which we aren't building, but it needs to exist to get past the ./configure step.

**Comment by Zishan-Rahman** [comment]
@libjared (sorry for the surprise ping) In the menatime, would you like to open a PR in [the Godot Engine Flatpak repository](https://github.com/flathub/org.godotengine.Godot) with your `libspeechd` module from here? If you did, it would address https://github.com/flathub/org.godotengine.Godot/issues/122, which is an issue that's been open for over a year now. Godot _isn't_ an Electron app so its Flatpak will greatly benefit from having `libspeechd` built and included alongside the main Godot executable.

File: dev.vencord.Vesktop.yml (hunk 3 above)
Comment (Vendicated) [reply dev.vencord.Vesktop.yml]: would be good to add a comment stating this

i got a bit confused by it, especially because it's such an unpopular repo

**Comment by Vendicated** [comment]
> I'll open a PR with them later, but if they turn me away I'm coming right back here :P  

have you done this yet? i checked your activity but couldn't find it

**Comment by hypevhs** [comment]
@Vendicated I opened the PR against the Electron BaseApp

**Comment by hypevhs** [comment]
bot, build

**Comment by hypevhs** [comment]
This^ build works and has TTS.

**Comment by Vendicated** [comment]
looks like your upstream pr got merged, awesome! can we just update the base now? or do we need to wait for them to make a new release

**Comment by Vendicated** [comment]
oh, nvm. awesome, thank you!!

will do some testing, then we can merge this

**Comment by hypevhs** [comment]
It took me some digging before, but I eventually found buried in the Flathub documentation that when new commits are added to an app's (or BaseApp's) branch, it explicitly waits exactly 3 hours before it updates the Flathub repos with the new build. I waited till after, and then asked the bot to rebuild.

TLDR AFAICT this is ready.

**Review by Vendicated — APPROVED** [review]
looks good

**Comment by Vendicated** [comment]
thanks for your work on this! :)
